Source/Core/DolphinQt/QtUtils/FileOpenEventFilter.cpp

```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <QFileOpenEvent>
+
+#include "DolphinQt/QtUtils/FileOpenEventFilter.h"
+
+FileOpenEventFilter::FileOpenEventFilter(QObject* event_source)
```

Comment:
This should probably be:

```suggestion
FileOpenEventFilter::FileOpenEventFilter(QObject* event_source) : QObject{event_source}
{
...
}
```

Allowing the event filter to automatically be deleted when the source gets deallocated, removing the need for the delete in MainWindow's destructor.

The way the current code is organized technically results in a dangling pointer to the event filter. When the main window begins being destructed, the application instance is still alive, but it has an event filter registered with it that hasn't been removed and has since been deallocated.